### ISSUE TITLE
SDDM doesn't quit the greeter gracefully with reuseSession=true

### ISSUE THREAD

**Issue by Vogtinator** [sddm/sddm#1282]
On successful logins, sddm doesn't tell the greeter to quit, it just kills the X server it is connected to while the greeter is about to quit already. This confuses it and it crashes.

**Comment by grahamperrin** [comment]
Please, is this still true? 

Maybe I'm missing something, but I don't see a crash on FreeBSD. 

```text
% pkg info -x sddm
plasma5-sddm-kcm-5.24.3
sddm-0.19.0_7
sddm-freebsd-black-theme-1.3
% uname -aKU
FreeBSD mowa219-gjp4-8570p-freebsd 14.0-CURRENT FreeBSD 14.0-CURRENT #8 main-n254268-50e244964e9-dirty: Mon Apr  4 09:29:49 BST 2022     root@mowa219-gjp4-8570p-freebsd:/usr/obj/usr/src/amd64.amd64/sys/GENERIC-NODEBUG amd64 1400056 1400056
% 
```

**Comment by Vogtinator** [comment]
The easiest way to reproduce it is to:
1. Log in
2. Lock the screen
3. Select "Switch user"
4. Log in as the already logged in user

SDDM then switches to the already existing session and abandons the greeter for the new session

**Comment by grahamperrin** [comment]
Thanks, I might have found alternative steps to reproduce: 

1. with ttyv8 (a norm on FreeBSD) in use by KDE Plasma
2. <kbd>Control</kbd>-<kbd>Alt</kbd>-<kbd>F2</kbd> to ttyv1, login as root, `tail -f -n 0 /var/log/messages`
3. <kbd>Control</kbd>-<kbd>Alt</kbd>-<kbd>F9</kbd> to ttyv8, take the menu option to switch user
5. observe the SDDM screen, **refrain from login**
6. <kbd>Control</kbd>-<kbd>Alt</kbd>-<kbd>F2</kbd> to ttyv1, observe the logged crash.

```text
% tail -n 2 /var/log/messages
Apr  6 00:22:41 mowa219-gjp4-8570p-freebsd kernel: pid 50425 (sddm-greeter), jid 0, uid 219: exited on signal 6 (core dumped)
Apr  6 00:23:19 mowa219-gjp4-8570p-freebsd kernel: pid 50473 (kcminit), jid 0, uid 1002: exited on signal 6 (core dumped)
% 
```

tl;dr (correct me, please, if I'm wrong) it seems that simply bringing up the screen (in response to _Switch User_ is enough to trigger a crash, at least on FreeBSD.

**Comment by Vogtinator** [comment]
It's a bit strange that `kcminit` also crashed. I guess that's unrelated?

**Comment by grahamperrin** [comment]
> I guess that's unrelated?

I guess so, looking at the time (long afterwards).

**Comment by cheongi** [comment]
> The easiest way to reproduce it is to:
> 
>     1. Log in
> 
>     2. Lock the screen
> 
>     3. Select "Switch user"
> 
>     4. Log in as the already logged in user

FreeBSD 13.1-RC2 can't replicate that atm. Locked screen disables "switch user". From locked screen, I can reuse session -- that is the only way I can get back in -- have to log out of other user sessions until I can magically find a lock screen.

I have tried to fix a typo bug, maybe that solved some issues.
[https://github.com/sddm/sddm/issues/1540](url)

Suspect there are bugs in the state transition handling. Presently I can't switch users at all -- only have 4 running for a day since testing yesterday.

**Comment by Vogtinator** [comment]
> > The easiest way to reproduce it is to:
> > ```
> > 1. Log in
> > 
> > 2. Lock the screen
> > 
> > 3. Select "Switch user"
> > 
> > 4. Log in as the already logged in user
> > ```
> 
> FreeBSD 13.1-RC2 can't replicate that atm. Locked screen disables "switch user". From locked screen, I can reuse session -- that is the only way I can get back in -- have to log out of other user sessions until I can magically find a lock screen.

Note that the session reuse/switch works fine, it's just that there's an ugly error in logs.

**Comment by cheongi** [comment]
sddm on freebsd with no logind doesn't appear to have any code to manage session reuse using console kit. So is someone responsible for that??

[https://github.com/sddm/sddm/blob/develop/src/daemon/Display.cpp](url)
